File: tests/Circuit.Interop.Tests/DocumentationExamples/Workflows.cs
using Circuit;

namespace Circuit.Interop.Tests.DocumentationExamples;

internal static class WorkflowExample
{
    public static WorkflowDefinition<int, bool> Build(AgentDefinition agent, AgentSignature<int, bool> signature)
        => WorkflowDefinition<int, bool>
            .Start("approval.workflow", "1.0.0", "classify", agent, signature)
            .RequestApproval(
                "manager.approval",
                approved => new ApprovalPrompt($"Approve {approved}", "A human must approve before completion."))
            .Then(
                "decision",
                (context, response, cancellationToken) => Task.FromResult(response.Approved))
            .Build();
}

File: tests/Circuit.FSharp.Tests/DocumentationExamples/Workflows.fs
namespace Circuit.FSharp.Tests.DocumentationExamples

open System.Threading
open System.Threading.Tasks
open Circuit.Core

module WorkflowsExample =
    let approvalFlow =
        Workflow.request "manager.approval" (fun approved ->
            ApprovalPrompt.Create($"Approve {approved}", "A human must approve before completion."))
        |> Workflow.define "approval.workflow" "1.0.0"
        |> Workflow.thenStep (Workflow.code "decision" (fun _ response -> Task.FromResult response.Approved))

    let start (runtime: IWorkflowRuntime) =
        Workflow.start runtime approvalFlow 42 WorkflowRunOptions.Default CancellationToken.None
